Support `ts-node` by checking `.ts` source files

marshall007 opened the pull request:
Should fix #41, but I haven't been able to test yet.

I still have a few more cases to update like in [`help.ts#L108-L117`](https://github.com/marshall007/clime/blob/marshall_41/src/core/command/help.ts#L108-L117), but I wanted to run this PR by you before continuing (mostly because I'm unsure how we should go about adding tests for this).

For what it's worth, I also don't think this is a good long-term solution. I would really like to see #43 implemented, but this seemed like the easier path forward for now to get my feet wet.

marshall007 commented:
Note the build failures already existed on `master` and are not a result of this change.

marshall007 commented:
@vilic I just pushed up another commit addressing the remaining cases in `help.ts`. Could still use some direction on testing. I can't really tell how the generator works. Perhaps it would be easier to just run `baseman` with `ts-node` prior to compiling the TS test source files?

vilic commented:
@marshall007 You might be right. We need some efforts on baseman implementation though. I will make necessary changes to baseman and test case generating here, and once done will merge your PR (if it passes). Thanks!

marshall007 commented:
@vilic no problem, just let me know if you need anything else! Cheers!

ajafff commented on src/internal-util/fs.ts:
```diff
@@ -11,6 +12,28 @@ export async function existsFile(path: string): Promise<boolean> {
   return !!stats && stats.isFile();
 }
 
+export async function existsSourceFile(path: string, filename: string = 'default'): Promise<string | undefined> {
```
why not simply use `require.resolve` to resolve a module with any supported file extension?

marshall007 commented:
@ajafff good call on using `require.resolve`, just updated this PR to do so. Note I rebased on top of 0e075f1 to get rid of those test failures. The existing (JS-based) tests still passed on my local using the new `require.resolve` logic.
